"Cannot directly print EncapsedStringPart" - How can I fix this?

error:
```
[ERROR] Could not process "/home/lmoelleken/testing/git/portable-utf8/src/voku/helper/UTF8.php" file, due to:          
         "Cannot directly print EncapsedStringPart". 
```

repo:
```git clone https://github.com/voku/portable-utf8```

command:
```vendor/bin/rector process src --level dead-code```

--- a/packages/CodingStyle/src/Rector/Use_/RemoveUnusedAliasRector.php
+++ b/packages/CodingStyle/src/Rector/Use_/RemoveUnusedAliasRector.php
@@ -89,7 +89,7 @@ CODE_SAMPLE
 
             // only alis name is used → use last name directly
             if (isset($usedNameNodes[$aliasName])) {
-                $this->renameNameNode($usedNameNodes, $aliasName, $lastName);
+                $this->renameNameNode($usedNameNodes[$aliasName], $lastName);
                 $use->alias = null;
             }
         }
@@ -98,7 +98,7 @@ CODE_SAMPLE
     }
 
     /**
-     * @return Name[][]|Node[][]
+     * @return Node[][][]
      */
     private function resolveUsedNameNodes(Node $parentNode): array
     {
@@ -122,10 +122,10 @@ CODE_SAMPLE
                     continue;
                 }
 
-                $usedNameNodes[$originalName->toString()][] = [
-                    $nameNode,
-                    $nameNode->getAttribute(Attribute::PARENT_NODE),
-                ];
+                /** @var Node $parentNode */
+                $parentNode = $nameNode->getAttribute(Attribute::PARENT_NODE);
+
+                $usedNameNodes[$originalName->toString()][] = [$nameNode, $parentNode];
             }
         }
 
@@ -133,11 +133,11 @@ CODE_SAMPLE
     }
 
     /**
-     * @param Name[][]|Node[][] $usedNameNodes
+     * @param Node[][] $usedNameNodes
      */
-    private function renameNameNode(array $usedNameNodes, string $aliasName, string $lastName): void
+    private function renameNameNode(array $usedNameNodes, string $lastName): void
     {
-        foreach ($usedNameNodes[$aliasName] as [$usedName, $parentNode]) {
+        foreach ($usedNameNodes as [$usedName, $parentNode]) {
             foreach ($this->getObjectPublicPropertyNames($parentNode) as $parentNodePropertyName) {
                 if ($parentNode->{$parentNodePropertyName} !== $usedName) {
                     continue;

--- a/src/PhpParser/Node/Maintainer/PropertyMaintainer.php
+++ b/src/PhpParser/Node/Maintainer/PropertyMaintainer.php
@@ -4,6 +4,7 @@ namespace Rector\PhpParser\Node\Maintainer;
 
 use PhpParser\Node;
 use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\StaticPropertyFetch;
 use PhpParser\Node\Stmt\Property;
 use Rector\NodeTypeResolver\Node\Attribute;
 use Rector\PhpParser\Node\BetterNodeFinder;
@@ -53,8 +54,8 @@ final class PropertyMaintainer
                 return null;
             }
 
-            // property
-            if (! $node instanceof PropertyFetch) {
+            // property + static fetch
+            if (! $node instanceof PropertyFetch && ! $node instanceof StaticPropertyFetch) {
                 return null;
             }
 

--- a/src/PhpParser/Printer/BetterStandardPrinter.php
+++ b/src/PhpParser/Printer/BetterStandardPrinter.php
@@ -15,6 +15,7 @@ use PhpParser\Node\Expr\StaticCall;
 use PhpParser\Node\Expr\Variable;
 use PhpParser\Node\Expr\Yield_;
 use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar\EncapsedStringPart;
 use PhpParser\Node\Scalar\String_;
 use PhpParser\Node\Stmt\Class_;
 use PhpParser\Node\Stmt\ClassMethod;
@@ -54,6 +55,10 @@ final class BetterStandardPrinter extends Standard
     {
         if ($node === null) {
             $node = [];
+        }
+
+        if ($node instanceof EncapsedStringPart) {
+            return 'UNABLE_TO_PRINT_ENCAPSED_STRING';
         }
 
         if (! is_array($node)) {

